src/components/TemplateList/TemplateItem.tsx

```diff
@@ -30,11 +36,28 @@ const TemplateItem = (prop: Props) => {
     designDetail: prop.template.design,
   };
 
-  const handleTemplateSelect = () => {
+  const handleCourtSelecting = (courtSizeName: string): void => {
+    dispatch(setActiveCourt(courtSizeName));
+    dispatch(updateBorderLength(templateItem.designDetail.courtSize.sideBorderWidth));
+    const selectedCourt = courtsData.find((item) => item.courtName === courtSizeName);
+    const tileQtyOfSelectedCourt = mockTileData.find(
+      (item) => item.name === selectedCourt?.courtName
+    )?.tileQty as AreaTileQty[];
+    dispatch(changeCourtType(tileQtyOfSelectedCourt));
+    dispatch(changeWholeCourtColor(templateItem.designDetail.tileColor));
+    dispatch(resetAll());
+  };
+
+  const selectTemplate = () => {
     setHoverOn(true);
     dispatch(startSelectTemplate(true));
```

Comment:
There seems no need to use "startSelectTemplate()", as "isTemplateSelect" is not used in other files.